alacritty/src/main.rs

```diff
@@ -213,8 +213,16 @@ fn run(config: Config, message_buffer: MessageBuffer) -> Result<(), Box<dyn Erro
 
     // Main display loop
     loop {
+        // Wait for events
+        processor.wait_for_events(display.window());
+
+        let mut terminal_lock = terminal.lock();
+
+        // Make sure that events are processed with current URLs
+        terminal_lock.update_urls();
+
         // Process input and window events
-        let mut terminal_lock = processor.process_events(&terminal, display.window());
+        processor.process_events(&mut terminal_lock, display.window());
```

Comment:
There have been a lot of changes to `main.rs` and `event.rs`, which would likely introduce a lot of troubles with the #2438 PR.

Looking at these changes, it seems like it might be possible to just move `update_urls()` inside of the `process_events` method and that would just be a single line change instead of the differences introduced here.

This was just assessed based on a quick read over the code without any testing however, so it's possible that this is incorrect. If that's the case, maybe it's better to base this PR on top of the glutin rework instead?

Reply:
OK, yeah, putting it in process_events makes sense if it makes the glutin rework easier - I made this change mostly just for ergonomics.